lib/shopify_api/resources/checkout.rb

```diff
@@ -8,5 +8,21 @@ class Checkout < Base
     def complete
       post(:complete)
     end
+
+    def ready?
+      return false unless persisted?
+
+      reload
```

Comment:
Why do you need to reload here?

Reply:
since the Checkout API must be polled to know if the checkout is ready (e.g. to retrieve shipping rates or before completing payment), we should reload the resource from the server to see what the status returned in the response is.

do you think it's better to have the client explicitly call `reload`?

Reply:
Actually, I think we _need_ to reload here since we are reading the status code from the response object. We don't actually store the status code anywhere on the checkout instance.